google/resource_sql_source_representation_instance.go

```diff
@@ -0,0 +1,227 @@
+package google
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"google.golang.org/api/googleapi"
+	sqladmin "google.golang.org/api/sqladmin/v1beta4"
+)
+
+func resourceSqlSourceRepresentationInstance() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceSqlSourceRepresentationInstanceCreate,
+		Read:   resourceSqlSourceRepresentationInstanceRead,
+		Delete: resourceSqlSourceRepresentationInstanceDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceSqlSourceRepresentationInstanceImporter,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"database_version": {
+				Type:     schema.TypeString,
+				Default:  "MYSQL_5_7",
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"host": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"port": {
+				Type:         schema.TypeInt,
+				Default:      3306,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IntBetween(0, 65535),
+			},
+
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"self_link": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceSqlSourceRepresentationInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	region, err := getRegion(d, config)
+	if err != nil {
+		return err
+	}
+
+	var name string
+	if v, ok := d.GetOk("name"); ok {
+		name = v.(string)
+	} else {
+		name = resource.UniqueId()
+	}
+	d.Set("name", name)
+
+	host := d.Get("host").(string)
+	port := d.Get("port").(int)
+	hostPort := fmt.Sprintf("%s:%d", host, port)
+
+	instance := &sqladmin.DatabaseInstance{
+		Name:            name,
+		Region:          region,
+		DatabaseVersion: d.Get("database_version").(string),
+		OnPremisesConfiguration: &sqladmin.OnPremisesConfiguration{
+			HostPort: hostPort,
+		},
+	}
+
+	backoff := time.Second
```

Comment:
Is there a reason you aren't using a retryTime() like in delete/other sql instances?

Reply:
For reference: [retryTime()](https://github.com/terraform-providers/terraform-provider-google/blob/1319ddd86de649e62984da8a31ca5643c4ef1787/google/utils.go#L322) which uses [isRetryableError()](https://github.com/terraform-providers/terraform-provider-google/blob/1319ddd86de649e62984da8a31ca5643c4ef1787/google/utils.go#L341)